FEAT: 스케쥴링 기능을 구현한다

SeongHo5356 opened the pull request:
## 💡 관련 이슈
- #39 

## 📝 작업 내용
- taskScheduler를 사용한다

## 🧑‍💻 변경 사항


## 💬 리뷰 요구사항(선택)

SeongHo5356 committed:
#37 Feat: 에러타입을 지정한다

SeongHo5356 committed:
#37 Feat: 작업을 예약한다.

SeongHo5356 committed:
#37 Feat: 작업 예약 관련 테스트코드를 작성한다.

SeongHo5356 committed:
#37 Chore: .gitignore을 최신화한다

yumzen commented on src/main/java/org/zapply/product/global/scheduler/service/ScheduledJobService.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Transactional
+    public void completeJob(Long jobId) {
+        jobRepository.findById(jobId).ifPresent(job -> {
```
jobId에 해당하는 Job이 없을 경우의 예외처리도 필요할 것 같아요!

SeongHo5356 replied:
수정했습니다~!

yumzen commented on src/main/java/org/zapply/product/global/config/SchedulerConfig.java:
```diff
@@ -0,0 +1,21 @@
+@Configuration
+public class SchedulerConfig {
+
+    private static final int POOL_SIZE = 3;
```
스레드풀을 3으로 설정해놓은 이유가 있나요??

SeongHo5356 replied:
이부분은 공부해보고 남겨놓을게요

yumzen reviewed (approved):
고생많으셨습니다! 많이 배워갑니다 :)

SeongHo5356 committed:
Merge branch 'develop' into feat/#37-scheduling

SeongHo5356 committed:
#37 Refactor: 예정된 작업을 찾을 수 없을 때 에러코드를 작성한다

# Conflicts:
#	src/main/java/org/zapply/product/global/apiPayload/exception/GlobalErrorType.java

SeongHo5356 committed:
#37 Refactor: CI/CD 과정을 분리한다

SeongHo5356 committed:
#37 Refactor: CI과정에서 테스트 제거한다

SeongHo5356 committed:
#37 Refactor: CI과정에서 환경변수를 포함시킨다
